system/jlib/jsecrets.cpp

```diff
@@ -368,17 +369,61 @@ static bool isEmptyTimeval(const timeval &tv)
 
 //---------------------------------------------------------------------------------------------------------------------
 
+//The secret key has the form category/name[@vaultId][#version]
+
+static std::string buildSecretKey(const char * category, const char * name, const char * optVaultId, const char * optVersion)
+{
+    std::string key;
+    key.append(category).append("/").append(name);
+    if (optVaultId)
+        key.append("@").append(optVaultId);
+    if (optVersion)
+        key.append("#").append(optVersion);
+    return key;
+}
+
+static void expandSecretKey(std::string & category, std::string & name, std::string & optVaultId, std::string & optVersion, const char * key)
+{
+    const char * slash = strchr(key, '/');
+    assertex(slash);
+    const char * at = strchr(slash, '@');
+    const char * hash = strchr(slash, '#');
+
+    const char * end = nullptr;
```

Comment:
Feels like this code would be simpler if you initialised end = key+strlen(key) rather than null

Reply:
It would be, not sure I want to change it no though.